lib/src/json_schema/json_schema.dart

```diff
@@ -466,6 +544,83 @@ class JsonSchema {
         'Data provided to createSubSchema is not valid: Must be a Map (or bool in draft6 or later). | ${schemaDefinition}');
   }
 
+  JsonSchema _fetchRefSchemaFromSyncProvider(Uri ref) {
+    // Always check refMap first.
+    if (_refMap.containsKey(ref.toString())) {
+      return _refMap[ref.toString()];
+    }
+
+    final Uri baseUri = ref.removeFragment();
+
+    // Fallback order for ref provider:
+    // 1. Full URI (example: localhost:1234/integer.json#/definitions/integer)
```

Comment:
Should we change the definition of RefProviders to say they should only provide root schemas?

Reply:
I like this idea yeah